### ISSUE TITLE
negative progress values

### ISSUE THREAD

**Issue by AlainBarthelemy** [Hemisphere-Project/HPlayer#15]
when sending s/getStatus command, sometimes the player's object returned contains progress with negative values. For instance :

```
 { name: 'Thierry',
  isPlaying: true,
  media: 
   { filepath: '/home/pi/media/Sequence 04 no bomb.mov',
     progress: -99291,
     duration: 450240 },
  isPaused: false,
  isMuted: false,
  volume: 50,
  zoom: 100,
  blur: 0,
  loop: true,
  hdmiAudio: true,
  info: false }
```

**Comment by jbum** [comment]
This is caused by 32-bit signed integer overflow in the multiply operation in this routine:

    int omPlayer::getPositionMs(){
        return static_cast<int>(this->getCurrentFrameNbr() * this->getDurationMs() / this->getTotalNumFrames());
    }

It can happen on even relatively short videos.  For example if DurationMS=639640 and currentFrameNumber=4200 the result is 2,686,488,000, which exceeds the max signed positive value for a 32-bit integer (2147483647).

It can be fixed by casting one of the multiplied vars to a double (or a 64-bit integer) to prevent the overflow.

e.g.

    int omPlayer::getPositionMs(){
         return static_cast<int>( (double) this->getCurrentFrameNbr() * this->getDurationMs() / this->getTotalNumFrames());
    }

**Comment by Maigre** [comment]
good point jbum, thanks !
i added the fix in commit 867fbd498a5b87a02924d36614be0c0b6088c3e3
need to be tested and validated, but that was obvisouly the problem.
:thumbsup:

**Comment by Maigre** [comment]
seems fixed now.
